tests/pathfinding/test_fee_schedule.py

```diff
@@ -125,36 +125,68 @@ def test_fees_in_balanced_routing():
     # When approximation iterations matter, those are given as sums of the steps.
 
     # Incoming channel
-    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(200))])
+    # Without fee capping
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(200))], cap_fees=False)
     assert tn.estimate_fee(1, 3) == 10 + 1
     assert tn.estimate_fee(3, 1) == -10
+    # With fee capping
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(200))], cap_fees=True)
+    assert tn.estimate_fee(1, 3) == 10 + 1
+    assert tn.estimate_fee(3, 1) == 0
 
     # The opposite fee schedule should give opposite results
-    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(200)), (TA(2000), FA(0))])
+    # Without fee capping
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(200)), (TA(2000), FA(0))], cap_fees=False)
     assert tn.estimate_fee(1, 3) == -10 + 1
     assert tn.estimate_fee(3, 1) == 10
+    # With fee capping
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(200)), (TA(2000), FA(0))], cap_fees=True)
+    assert tn.estimate_fee(1, 3) == 0
+    assert tn.estimate_fee(3, 1) == 10
 
     # Outgoing channel
     tn.set_fee(2, 1)
-    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(200))])
+    # Without fee capping
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(200))], cap_fees=False)
     assert tn.estimate_fee(1, 3) == -10
     assert tn.estimate_fee(3, 1) == 10 + 1
+    # With fee capping
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(200))], cap_fees=True)
+    assert tn.estimate_fee(1, 3) == 0
+    assert tn.estimate_fee(3, 1) == 10 + 1
 
     # The opposite fee schedule should give opposite results
-    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(200)), (TA(2000), FA(0))])
+    # Without fee capping
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(200)), (TA(2000), FA(0))], cap_fees=False)
     assert tn.estimate_fee(1, 3) == 10
     assert tn.estimate_fee(3, 1) == -10 + 1
+    # With fee capping
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(200)), (TA(2000), FA(0))], cap_fees=True)
+    assert tn.estimate_fee(1, 3) == 10
+    assert tn.estimate_fee(3, 1) == 0
 
     # Combined fees cancel out
-    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(20))])
-    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(20))])
+    # Only works without fee capping
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(20))], cap_fees=False)
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(20))], cap_fees=False)
     assert tn.estimate_fee(1, 3) == 0
     assert tn.estimate_fee(3, 1) == 0
+    # With fee capping there is a fee
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(2000), FA(20))], cap_fees=True)
```

Comment:
Mh, I guess this should work with capped fees as well.

Reply:
Yes, I think so, too.